contracts/sdk/AQueryProofVerifierBuilder.sol

```diff
@@ -0,0 +1,167 @@
+     *      based on application-specific data encoded in `userPayload_`.
+     * @param userPayload_ Encoded application-specific data passed from the external `execute` or `executeNoir` call.
+     */
+    // solhint-disable-next-line no-empty-blocks
+    function _afterVerify(bytes memory userPayload_) public virtual {}
+
+    /**
+     * @notice Abstract function responsible for constructing the public signals array for the ZK proof.
+     * @dev This function should decode `userPayload_` and use the `PublicSignalsBuilder` library functions
+     * (attached via `using for`) to populate the public signals array based on the specific application's requirements.
+     * @param userPayload_ Encoded application-specific data required to build the public signals.
+     * @return dataPointer_ A `uint256` representing the memory pointer to the constructed public signals array.
+     */
+    function _buildPublicSignals(
+        bytes memory userPayload_
+    ) public virtual returns (uint256 dataPointer_);
+
+    /**
+     * @notice Executes the full ZK proof verification workflow for a Circom (Groth16) proof.
+     * @param registrationRoot_ The root of the identity SMT against which the proof was generated.
+     * @param currentDate_ The current date (encoded as `yyMMdd`) to be included in the public signals.
+     * @param userPayload_ Encoded application-specific data to be used by hooks and the signal builder.
+     * @param zkPoints_ The Circom Groth16 proof points (`ProofPoints` struct).
+     */
+    function execute(
+        bytes32 registrationRoot_,
+        uint256 currentDate_,
+        bytes memory userPayload_,
+        ProofPoints memory zkPoints_
+    ) external {
+        _beforeVerify(userPayload_);
+
+        uint256 builder_ = _buildPublicSignals(userPayload_);
+        builder_.withCurrentDate(currentDate_);
+        builder_.withIdStateRoot(registrationRoot_);
+
+        uint256[] memory publicSignals_ = PublicSignalsBuilder.buildAsUintArray(builder_);
+
+        if (!_verifyCircomProof(zkPoints_, publicSignals_)) {
+            revert InvalidCircomProof(publicSignals_, zkPoints_);
+        }
+
+        _afterVerify(userPayload_);
+    }
+
+    /**
+     * @notice Executes the full ZK proof verification workflow for a Noir proof.
+     * @param registrationRoot_ The root of the identity SMT against which the proof was generated.
+     * @param currentDate_ The current date (encoded as `yyMMdd`) to be included in the public signals.
+     * @param userPayload_ Encoded application-specific data to be used by hooks and the signal builder.
+     * @param zkPoints_ The raw bytes of the Noir proof.
+     */
+    function executeNoir(
+        bytes32 registrationRoot_,
+        uint256 currentDate_,
+        bytes memory userPayload_,
+        bytes memory zkPoints_
+    ) external {
+        _beforeVerify(userPayload_);
```

Comment:
Let's pass `registrationRoot_` and `currentDate_` to hooks as well. Like, why not?